### PR TITLE
Mm 30317 add new `manage remote clusters` permission

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-30317

Updated:
- Self-Managed Admin Guide > Configure Mattermost > Mattermost Settings > Experimental Settings only in config.json
   - Added new **Remote Clusters** config setting
- Self-Managed Admin Guide > Onboard Users > Advanced Permissions: Backend Infrastructure > Data Structure > Permissions
   - Added new **manage_remote_clusters** system permission to the table

### CODE REVIEW

File: source/administration/config-settings.rst [hunk 1]
```diff
@@ -4714,6 +4714,19 @@ Audit settings
...
+
+*Available in Enterprise Edition E20*
+
+Enable this setting to add, remove, and view remote clusters for shared channels.
+
+**True**: When ``true`` System Admins can manage remote clusters using the System Console.
+
+**False**: Remote cluster management is disabled.
+
+This feature's ``config.json`` setting is ``"RemoteClusters": false`` with options ``true`` and ``false``.
```
Comment (amyblais) [inline source/administration/config-settings.rst:4728]: The setting should be formatted in a table format, please see the settings below it for examples.

Comment (cwarnermm) [reply source/administration/config-settings.rst]: Thanks, @amyblais!

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

File: source/administration/config-settings.rst [hunk 2]
```diff
@@ -4714,6 +4714,21 @@ Audit settings
 
 The audit settings output audit records to syslog (local or remote server via TLS) and/or to a local file. Both are disabled by default. They can be enabled simultaneously.
 
+Remote Clusters
```
Comment (amyblais) [inline source/administration/config-settings.rst:4717]: I believe the setting can also be documented in the telemetry doc.

Comment (cwarnermm) [reply source/administration/config-settings.rst:4717]: Thanks, @amyblais! I've added 'bool RemoteClusters' to the Telemetry doc.

Comment (wiersgallak) [inline source/administration/config-settings.rst:4717]: @cwarnermm Just confirming this is under the experimental section - I believe so but may have missed the heading in my scrolling.

Comment (cwarnermm) [reply source/administration/config-settings.rst:4717]: @wiersgallak Confirmed. This new config setting is documented under the "Experimental Settings only in config.json" section.

**Review by justinegeffen — APPROVED** [review]

**Review by wiersgallak — APPROVED** [review]
Thanks!

**Pull request merged**